account_banking_pain_base/models/account_payment_order.py

```diff
@@ -50,31 +31,14 @@ class AccountPaymentOrder(models.Model):
         "false, the bank statement will display one debit line per wire "
         "transfer of the SEPA XML file.")
 
-    @api.multi
-    @api.depends(
-        'company_partner_bank_id.acc_type',
-        'payment_line_ids.currency_id',
-        'payment_line_ids.partner_bank_id.acc_type')
-    def compute_sepa(self):
-        eur = self.env.ref('base.EUR')
-        for order in self:
-            sepa = True
-            if order.company_partner_bank_id.acc_type != 'iban':
-                sepa = False
-            for pline in order.payment_line_ids:
-                if pline.currency_id != eur:
-                    sepa = False
-                    break
-                if pline.partner_bank_id.acc_type != 'iban':
-                    sepa = False
-                    break
-            sepa = order.compute_sepa_final_hook(sepa)
-            self.sepa = sepa
-
-    @api.multi
-    def compute_sepa_final_hook(self, sepa):
-        self.ensure_one()
-        return sepa
+    @api.model
+    def _prepare_bank_payment_line(self, paylines):
+        result = super()._prepare_bank_payment_line(paylines)
+        result.update({
+            'sepa': paylines[0].sepa,
+            'charge_bearer': paylines[0].charge_bearer
+        })
```

Comment:
A comment that explain why it is safe to use `paylines[0]` would be helpful (do we have guarantee that all payline have the same value for these two fields?).

Reply:
BTW, why is there a need to write the `sepa` field when it is actually a computed field?

Reply:
Indeed, sepa is not needed anymore. I corrected it in my last commit (https://github.com/OCA/bank-payment/pull/658/commits/1b5fd8797020563bc7b9c35a5539333631874a28).

Paylines[0] is just fine because paylines are now grouped also by sepa and charge_bearer values so all paylines have the same charge_bearer (https://github.com/OCA/bank-payment/pull/658/commits/2c14f5d7b7252a670b78f78c25b3c4f850e39405#diff-572f2cb9ee06aba671bb3532e23ca120R43).